pkg/scheduler/scheduler.go

```diff
@@ -134,14 +135,24 @@ func (s *Scheduler) schedule(ctx context.Context) {
 
 	// 6. Requeue the heads that were not scheduled.
 	for _, e := range entries {
-		if e.admissible {
-			continue
+		if e.status != admitted {
+			s.requeueAndUpdate(log, ctx, &e.Info, e.inadmissibleReason)
+			s.recorder.Eventf(e.Obj, corev1.EventTypeNormal, "Pending", e.inadmissibleReason)
 		}
-		s.requeueAndUpdate(log, ctx, &e.Info, e.noAdmissionReason)
-		s.recorder.Eventf(e.Obj, corev1.EventTypeNormal, "Pending", e.noAdmissionReason)
 	}
 }
 
+type entryStatus string
+
+const (
+	// indicates if the workload is nominated for admission.
+	nominated entryStatus = "nominated"
+	// indicates if the workload was nominated but skipped in this cycle.
+	skipped entryStatus = "skipped"
+	// indicates if the workload was successfully admitted.
+	admitted entryStatus = "admitted"
```

Comment:
`assumed` is more accurate, because the API call is async.

but that's a nit. I'm ok with this too.